src/python/pants/backend/jvm/tasks/jvm_compile/jvm_compile.py

```diff
@@ -716,6 +725,10 @@ def work_for_vts(vts, ctx):
                             compiler_option_sets,
                             zinc_file_manager,
                             counter)
+
+        # This might be missing the analysis file?
+        ctx.classes_dir = ClasspathEntry(ctx.classes_dir.path, directory_digest)
```

Comment:
@illicitonion: From what I can tell by following the code through, this won't contain the analysis file. Is that an issue? Wouldn't my test have failed if it was?

Reply:
If the analysis file is missing, zinc will execute a clean compile (ie, it will not execute an incremental compile), so it will still succeed.

Honestly, it would probably be totally reasonable to require `--no-compile-zinc-incremental` when the hermetic strategy is in use (for now, with TODOs), and then never include analysis in input snapshots (to force them to always be clean compiles). You would still capture output analysis from a compile, and you'd still check it on disk in the relevant place, because that would put them in the build cache.

Reply:
We currently capture the analysis cache in the snapshot; see src/python/pants/backend/jvm/tasks/jvm_compile/zinc/zinc_compile.py line 434

@stuhood raises a reasonable point that we should probably either never capture the analysis file, or only capture it if the flag is set to say we should, though.

(It's also super hacky that we capture the analysis file in the same snapshot as the classpath entry, but it's a lot more convenient than trying to capture it in a separate snapshot)